lib/core/src/sdk.rs

```diff
@@ -1231,14 +1302,19 @@ impl LiquidSdk {
             .build_tx_or_drain_tx(
                 self.config.lowball_fee_rate_msat_per_vbyte(),
                 &address_data.address,
+                &asset_id,
                 receiver_amount_sat,
             )
             .await?;
         let tx_fees_sat = tx.all_fees().values().sum::<u64>();
         ensure_sdk!(tx_fees_sat <= fees_sat, PaymentError::InvalidOrExpiredFees);
 
         let tx_id = tx.txid().to_string();
-        let payer_amount_sat = receiver_amount_sat + tx_fees_sat;
+        let amount_sat = if asset_id.eq(LIQUID_BTC_ASSET_ID.as_str()) {
+            receiver_amount_sat + tx_fees_sat
+        } else {
+            receiver_amount_sat
```

Comment:
This seems inconsistent on how the sdk user needs to handle the amount.
In btc the amount is including the fees and other assets it it excluded. I think we should be consistent here.

Reply:
I think to do this we need an extra column PaymentTxData to store the btc amount and asset amount. Also in the Payment to have a amount_sat and amount_asset